File: Try_GameEngine5/src/com/example/try_gameengine/Camera/ACamera.java
package com.example.try_gameengine.Camera;

/**
 * Camera is common object in game engine. It can control the whole display. 
 * @author irons
 *
 */
public abstract class ACamera {
	
	/**
	 * rotation camera.
	 * @param rotation
	 */
	public abstract void rotation(float rotation);
	/**
	 * translate camera.
	 * @param dx for x-dir translate.
	 * @param dy for y-dir translate.
	 */
	public abstract void translate(float dx, float dy);
	/**
	 * zoom(change xscale or yscale) cameara.
	 * @param scale
	 */
	public abstract void zoom(float scale);
	
	public abstract void bindLayerXY();
	public abstract void bindLayerX();
	public abstract void bindLayerY();
	public abstract void setIsAutoStopOnBound();
}

File: Try_GameEngine5/src/com/example/try_gameengine/Camera/ACamera.kt
package com.example.try_gameengine.Camera

/**
 * Camera is common object in game engine. It can control the whole display.
 * @author irons
 // */
abstract class ACamera {
    /**
     * rotation camera.
     * @param rotation
     // */
    abstract fun rotation(rotation: Float)

    /**
     * translate camera.
     * @param dx for x-dir translate.
     * @param dy for y-dir translate.
     // */
    abstract fun translate(dx: Float, dy: Float)

    /**
     * zoom(change xscale or yscale) cameara.
     * @param scale
     // */
    abstract fun zoom(scale: Float)

    abstract fun bindLayerXY()
    abstract fun bindLayerX()
    abstract fun bindLayerY()
    abstract fun setIsAutoStopOnBound()
}
